### DISCUSSION TITLE
Storing site settings (including theme settings like images)

### DISCUSSION THREAD

**Discussion by joshwegener** [Laravel-Backpack/community-forum#877] [Q&A (Help)] [answered]
Hello,

I'm working on my first backpack website! I'm trying to figure out what is the "best" method for storing data like "logo", main background image, copyright and so on. I looked into the add-on settings, but that only allows you to store strings, not things like uploading an image.

My current solution is to just create a new model called site_theme_settings and add the need columns and only have 1 row. But this seems like a bad practice. So far I have just disabled CREATE and DELETE so you can just list / view / update the one row.

I'm using the PRO version and have DevTools. I also have the following add-ons: PageManager, PermissionManager, PageManager.

Thanks!

**Comment by jcastroa87** [comment, marked as answer]
Hello @joshwegener 

I think Setting is your best choice. Maybe you miss how to config fields, but there is a way to handle images, per example like this:

```
{"name":"value","label":"Value","type":"image"}
```

Then, this will store the image in base64 in db.

Cheers.

**Comment by joshwegener** [comment]
Hello @jcastroa87,

That would make all the settings “image” type. I need some to be text, some to be images. Something like, name, type, value where value would change based on what type you selected.

Thanks

**Comment by o15a3d4l11s2** [comment]
Just as @jcastroa87 explained, you can use the [settings package](https://github.com/Laravel-Backpack/Settings).
As an example here is what I seed into my database in order to have various types of fields
```
[
    'key' => 'general.site_title',
    'name' => __('backend.site_title'),
    'value' => '',
    'field' => '{"name":"value","label":"' . __('backend.value') . '","type":"text"}',
    'active' => 1,
], [
    'key' => 'general.product_reviews.enabled',
    'name' => __('backend.product_reviews'),
    'value' => false,
    'field' => '{"name":"value","label":"' . __('backend.product_reviews.enabled') . '","type":"checkbox"}',
    'active' => 1,
]
```
I have different fields definitions like `text`, `wysiwyg`, `checkbox`, `textarea`, `url`, etc. This way I can easily manage different types of fields that store different data. It is quite easy to do so.

You can add fields in a class
```
class SettingSeeder extends Seeder
{
    /**
     * Run the database seeds.
     *
     * @return void
     */
    public function run()
    {
        $settings = [
            [
                'key' => 'general.site_title',
                'name' => __('backend.site_title'),
                'value' => '',
                'field' => '{"name":"value","label":"' . __('backend.value') . '","type":"text"}',
                'active' => 1,
            ], [
                'key' => 'general.product_reviews.enabled',
                'name' => __('backend.product_reviews'),
                'value' => false,
                'field' => '{"name":"value","label":"' . __('backend.product_reviews.enabled') . '","type":"checkbox"}',
                'active' => 1,
            ],
            // Add more settings definitions
        ];

        foreach ($settings as $setting) {
            Setting::firstOrCreate([
                'key' => $setting['key']
            ], $setting);
        }
    }
}
```
and then when you need, you can run `php artisan db:seed --class=SettingSeeder`

**Reply by joshwegener** [reply]
Thank you, I should have read the entire README… thank you for explaining!

**Comment by jcastroa87** [comment]
Great, you solve it, I will close the issue, but feel free to re-open or create a new one.

Cheers.
